.github/workflows/tests.yaml

```diff
@@ -82,21 +82,26 @@ jobs:
           ~/.composer/vendor/bin/php-coveralls -v
 
   functional-tests:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     env:
       SELENIUM_SERVER_DOWNLOAD_URL: http://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar
 
     strategy:
       fail-fast: false
       matrix:
+        os: ['ubuntu-latest']
         browser: ['chrome', 'firefox']
+        experimental: [false]
         selenium-server: [true, false] # Whether to run via Selenium server or directly via browser driver
         w3c: [true] # Although all builds negotiate protocol by default, it implies W3C protocol for both Chromedriver and Geckodriver
         include:
           # Force OSS (JsonWire) protocol on ChromeDriver - to make sure we keep compatibility
-          - { browser: 'chrome', selenium-server: false, w3c: false }
+          - { browser: 'chrome', os: 'ubuntu-latest', selenium-server: false, w3c: false, experimental: false }
+          - { browser: 'safari', os: 'macos-latest', selenium-server: false, w3c: true, experimental: true }
+
+    continue-on-error: ${{ matrix.experimental }}
```

Comment:
I'm still a bit confused why this is needed. The test should be passing with all the exclude-group, right? So why then is this "experimental" flag needed?

Reply:
removed